### ISSUE TITLE
Clang format: Dry-run fails without expected changes

### ISSUE THREAD

**Issue by JVApen** [llvm/llvm-project#61048]
The following code keeps giving failures, even after reformatting.
Found on 16-rc2, looks like a possible release blocker?
````
class C
{
   C(size_t nr = 1'000) {}
};
````

Failure message:
````
test.cpp:3:18: warning: code should be clang-formatted [-Wclang-format-violations]
   C(size_t nr = 1'000) {}
                 ^
````

[reproduction.zip](https://github.com/llvm/llvm-project/files/10850414/reproduction.zip)

**Comment by JVApen** [comment]
Possible duplicate of https://github.com/llvm/llvm-project/issues/60948

**Comment by owenca** [comment]
I can't reproduce it:
```
C:\Users\Owen\llvm-project\build_rel>cat run.bat
bin\clang-format.exe --dry-run test.cpp

bin\clang-format.exe -i test.cpp

bin\clang-format.exe --dry-run test.cpp

C:\Users\Owen\llvm-project\build_rel>cat test.cpp
class C
{
   C(size_t nr = 1'000) {}
};

C:\Users\Owen\llvm-project\build_rel>cat .clang-format
Language:        Cpp
IntegerLiteralSeparator:
  Binary:          4
  Decimal:         3
  Hex:             4
C:\Users\Owen\llvm-project\build_rel>bin\clang-format -version
clang-format version 17.0.0 (https://github.com/llvm/llvm-project 3b090ff2bdb2828b53d6fec9c06b9d52fed09f42)

C:\Users\Owen\llvm-project\build_rel>run.bat

C:\Users\Owen\llvm-project\build_rel>bin\clang-format.exe --dry-run test.cpp

C:\Users\Owen\llvm-project\build_rel>bin\clang-format.exe -i test.cpp

C:\Users\Owen\llvm-project\build_rel>bin\clang-format.exe --dry-run test.cpp

C:\Users\Owen\llvm-project\build_rel>
```

**Comment by rymiel** [comment]
This was fixed by b98d95984f0eb81a5f06eeaa3df29bbb1834a04b (verified by bisect)
